### DISCUSSION TITLE
Change window duration

### DISCUSSION THREAD

**Discussion by RoelVugts** [pyannote/pyannote-audio#1568] [Q&A]
In the diarization pipeline is the following parameter:
`    segmentation_step: float, optional
        The segmentation model is applied on a window sliding over the whole audio file.
        "segmentation_step" controls the step of this window, provided as a ratio of its
        duration. Defaults to 0.1 (i.e. 90% overlap between two consecuive windows).`

Now changing this ratio speeds up the diarization process significantly (and without much loss of accuracy). Since the segmentation step is a ratio of the duration of the window. Is it also possible to extend the window length? I would like to experiment with longer window lengths to check the results compared with processing speed. Thanks!

**Comment by hbredin** [comment]
I did not expose this `duration` parameter because the segmentation model has only ever seen this duration during the training process so there is no garantee that it will behave nicely with other duration.

You can definitely try that though (and report back :pray:) but it would need a bit of work on your side to [edit the code](https://github.com/pyannote/pyannote-audio/blob/28b5531cec35c70b7f0353a502061dcd1dd11e1d/pyannote/audio/pipelines/speaker_diarization.py#L142).
